From: Jason Dillaman <dillaman@redhat.com>
Date: Thu, 15 Sep 2016 18:14:15 -0400
Subject: rbd-mirror: concurrent access of event might result in heap
 corruption

Fixes: http://tracker.ceph.com/issues/17283
Signed-off-by: Jason Dillaman <dillaman@redhat.com>
(cherry picked from commit ac9ad37ab80db8913f9f1149707bd0009f8235c4)

Resolves: rhbz#1349955
---
 src/tools/rbd_mirror/ImageReplayer.cc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/tools/rbd_mirror/ImageReplayer.cc b/src/tools/rbd_mirror/ImageReplayer.cc
index 565dced..7a206db 100644
--- a/src/tools/rbd_mirror/ImageReplayer.cc
+++ b/src/tools/rbd_mirror/ImageReplayer.cc
@@ -1043,7 +1043,6 @@ void ImageReplayer<I>::process_entry() {
     ImageReplayer, &ImageReplayer<I>::handle_process_entry_ready>(this);
   Context *on_commit = new C_ReplayCommitted(this, std::move(m_replay_entry));
   m_local_replay->process(m_event_entry, on_ready, on_commit);
-  m_event_entry = {};
 }
 
 template <typename I>
